blockchain/filedao/filedao.go

```diff
@@ -248,30 +248,102 @@ func (fd *fileDAO) PutBlock(ctx context.Context, blk *block.Block) error {
 		log.L().Error("Block already exists.", zap.Uint64("height", blk.Height()), log.Hex("hash", h[:]))
 		return ErrAlreadyExist
 	}
-	// TODO: check if need to split DB
+
+	// check if we need to split DB
+	if fd.cfg.SplitDBSize() > 0 {
+		if err := fd.prepNextDbFile(blk.Height()); err != nil {
+			return err
+		}
+	}
 	return fd.currFd.PutBlock(ctx, blk)
 }
 
+func (fd *fileDAO) prepNextDbFile(height uint64) error {
+	fd.lock.Lock()
+	defer fd.lock.Unlock()
+
+	tip, err := fd.currFd.Height()
+	if err != nil {
+		return err
+	}
+	if height != tip+1 {
+		return ErrInvalidTipHeight
+	}
+
+	split, err := fd.needToSplitDB(height)
+	if err != nil || !split {
+		return err
+	}
+	return fd.addNewV2File(height)
+}
+
+func (fd *fileDAO) needToSplitDB(height uint64) (bool, error) {
+	filename := fd.cfg.DbPath
+	if fd.topIndex > 0 {
+		filename = kthAuxFileName(fd.cfg.DbPath, fd.topIndex)
+	}
+
+	dat, err := os.Stat(filename)
+	if err != nil {
+		return false, err
+	}
+	return uint64(dat.Size()) > fd.cfg.SplitDBSize(), nil
+}
```

Comment:
I think there are better approach for this. it adds additional disk io to check file status for every put which is quite unnecessary

Reply:
changed to count blocks